src/utils.py

```diff
@@ -209,6 +212,41 @@ def flatten(nested_iterable: Iterable[Iterable[T]]) -> chain[T]:
     return chain.from_iterable(nested_iterable)
 
 
+def subprocess_args():
+    """
+    See https://github.com/pyinstaller/pyinstaller/wiki/Recipe-subprocess
+    for reference and comments.
+
+    This code snippet was copied from https://github.com/madmaze/pytesseract
+    """
+    kwargs = {
+        "stdin": subprocess.PIPE,
+        "stdout": subprocess.PIPE,
+        "stderr": subprocess.DEVNULL,
+        "startupinfo": None,
+        "env": environ,
+    }
+
+    if hasattr(subprocess, "STARTUPINFO"):
+        kwargs["startupinfo"] = subprocess.STARTUPINFO()
+        kwargs["startupinfo"].dwFlags |= subprocess.STARTF_USESHOWWINDOW
+        kwargs["startupinfo"].wShowWindow = subprocess.SW_HIDE
+
+    return kwargs
+
+
+def run_tesseract(png: bytes):
+    """
+    Executes the tesseract CLI and pipes a PNG encoded image to it.
+
+    @param png: PNG encoded image as byte array
+    @return: The recognized output string from tesseract
+    """
+    p = subprocess.Popen(TESSERACT_CMD, **subprocess_args())  # noqa: S603
```

Comment:
```suggestion
class SubprocessKWArgs(TypedDict): # TODO: move me to the top of the file with other utility classes, right under enums
    stdin: int
    stdout: int
    stderr: int
    startupinfo: subprocess.STARTUPINFO | None
    env: os._Environ[str] | None  # pyright: ignore[reportPrivateUsage]


def subprocess_kwargs():
    """
    Create a set of arguments which make a ``subprocess.Popen`` (and
    variants) call work with or without Pyinstaller, ``--noconsole`` or
    not, on Windows and Linux.

    Typical use::

      subprocess.call(['program_to_run', 'arg_1'], **subprocess_args())

    ---
    Originally found in https://github.com/madmaze/pytesseract/blob/master/pytesseract/pytesseract.py
    Recipe from https://github.com/pyinstaller/pyinstaller/wiki/Recipe-subprocess
    which itself is taken from https://github.com/bjones1/enki/blob/master/enki/lib/get_console_output.py
    """
    # The following is true only on Windows.
    if hasattr(subprocess, "STARTUPINFO"):
        # On Windows, subprocess calls will pop up a command window by default when run from
        # Pyinstaller with the ``--noconsole`` option. Avoid this distraction.
        startupinfo = subprocess.STARTUPINFO()
        startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
        # https://github.com/madmaze/pytesseract/blob/88839f03590578a10e806a5244704437c9d477da/pytesseract/pytesseract.py#L236
        startupinfo.wShowWindow = subprocess.SW_HIDE
        # Windows doesn't search the path by default. Pass it an environment so it will.
        env = os.environ
    else:
        startupinfo = None
        env = None

    # On Windows, running this from the binary produced by Pyinstaller
    # with the ``--noconsole`` option requires redirecting everything
    # (stdin, stdout, stderr) to avoid an OSError exception
    # "[Error 6] the handle is invalid."
    return SubprocessKWArgs(
        stdin=subprocess.PIPE,
        stdout=subprocess.PIPE,
        stderr=subprocess.DEVNULL,
        startupinfo=startupinfo,
        env=env,
    )


def run_tesseract(png: bytes):
    """
    Executes the tesseract CLI and pipes a PNG encoded image to it.

    @param png: PNG encoded image as byte array
    @return: The recognized output string from tesseract

    ---
    """
    return (
        subprocess
        .Popen(TESSERACT_CMD, **subprocess_kwargs())  # noqa: S603 # Only using known literal strings
        .communicate(input=png)[0]
        .decode()
    )
```
A couple requested changes at once:
1. I know the recipe calls it `subprocess_args`, but they're kwargs (I could suggest this change upstream)
2. Either you need to declare your dict as potentially having a `STARTUPINFO`: `kwargs: dict[str, int | subprocess.STARTUPINFO | _Environ[str] | None]` or you reorganize to use type inference like I did above and like the recipe does.
3. I still want comments explaining what's happening.
4. Trying to untangle the history of copy of this method
5. Explaining the noqa
6. Fix type-checking issues